packages/nodes-base/nodes/Mattermost/v1/actions/user/create.ts

```diff
@@ -0,0 +1,50 @@
+import {
+	IExecuteFunctions,
+} from 'n8n-core';
+
+import {
+	IDataObject,
+	INodeExecutionData,
+} from 'n8n-workflow';
+
+import {
+	apiRequest,
+} from '../../transport';
+
+export async function create(this: IExecuteFunctions, index: number): Promise<INodeExecutionData[]> {
+	const username = this.getNodeParameter('username', index) as string;
+	const authService = this.getNodeParameter('authService', index) as string;
+	const additionalFields = this.getNodeParameter('additionalFields', index) as IDataObject;
+	
+	const qs = {} as IDataObject;
+	const requestMethod = 'POST';
+	const endpoint = 'users';
+	const body = {} as IDataObject;
+
+	body.auth_service = authService;
+
+	body.username = username;
+	Object.assign(body, additionalFields);
+
+	if (body.notificationUi) {
+		body.notify_props = (body.notificationUi as IDataObject).notificationValues;
+	}
+
+	if (authService === 'email') {
+		body.email = this.getNodeParameter('email', index) as string;
+		body.password = this.getNodeParameter('password', index) as string;
+	} else {
+		body.auth_data = this.getNodeParameter('authData', index) as string;
+	}
+
+	const returnData: IDataObject[] = [];
+	const responseData = await apiRequest.call(this, requestMethod, endpoint, body, qs);
+
+	if (Array.isArray(responseData)) {
+		returnData.push(...responseData);
+	} else {
+		returnData.push(responseData);
+	}
+
+	return this.helpers.returnJsonArray(returnData);
```

Comment:
Before, we had this pattern once per node - now it will multiply by the number of node actions. Therefore, can we tuck away this logic inside one of the helpers? Same comment applies to array-ifying the output at the end of every router.